backend/qemu.pm

```diff
@@ -218,19 +218,27 @@ sub save_memory_dump {
 
     die(sprintf("Migration failed: desc: %s, class: %s, stopped", $rsp->{error}->{desc}, $rsp->{error}->{class})) if ($rsp->{error});
 
-
+    my $migration_starttime = gettimeofday;
+    my $execution_time      = gettimeofday;
     do {
 
         sleep 0.5;    #We want to wait a decent amount of time, a file of 1GB will be
                       # migrated in about 40secs with an ssd drive. and no heavy load.
+        $execution_time = gettimeofday - $migration_starttime;
         $rsp = $self->handle_qmp_command({execute => "query-migrate"});
 
         diag "Migrating total bytes:     \t" . $rsp->{return}->{ram}->{total};
         diag "Migrating remaining bytes:   \t" . $rsp->{return}->{ram}->{remaining};
+        # 240 seconds should be ok for 4GB
+        if ($execution_time > 240) {
```

Comment:
I don't like magic numbers, but :/

Reply:
better extract it into a constant with a helpful name

Reply:
I suggest a $wtf_why_is_this_still_not_done = 240

Reply:
worksforme :)

Reply:
set it to $max_execution_time :)